File: 0841-keys-and-rooms/0841-keys-and-rooms.py
class Solution:
    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        vis = [False for i in range(len(rooms))]
        q = deque()
        for i in rooms[0]:
            q.append(i)
        vis[0] = True
        while q:
            t = q.popleft()
            if not vis[t]:
                for j in rooms[t]:
                    if not vis[j]: q.append(j)
            vis[t] = True
        res = True
        for i in vis:
            res &= i
        return res

File: 0841-keys-and-rooms/0841-keys-and-rooms.java
class Solution {
    public boolean canVisitAllRooms(List<List<Integer>> rooms) {
        boolean[] vis = new boolean[rooms.size()];
        Queue<Integer> queue = new LinkedList<>();
        for(int i : rooms.get(0))
            queue.offer(i);
        vis[0] = true;
        while(!queue.isEmpty()) {
            int t = queue.poll();
            if(!vis[t]) {
                for(int i : rooms.get(t)) {
                    if(!vis[i]) queue.offer(i);
                }
            }
            vis[t] = true;
        }
        boolean ans = true;
        for(boolean b : vis) ans &= b;
        return ans;
    }
}